sdk/objc/api/peerconnection/RTCFrameCryptorKeyManager.mm

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2022 LiveKit
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "RTCFrameCryptorKeyManager+Private.h"
+
+#include <memory>
+
+#import "base/RTCLogging.h"
+#import "helpers/NSString+StdString.h"
+
+class DefaultKeyManagerImpl : public webrtc::KeyManager {
+ public:
+  DefaultKeyManagerImpl() = default;
+  ~DefaultKeyManagerImpl() override = default;
+
+  /// Set the key at the given index.
+  bool SetKey(const std::string participant_id, int index, std::vector<uint8_t> key) {
+    if (index > webrtc::KeyManager::kMaxKeySize) {
+      return false;
+    }
+
+    if (keys_.find(participant_id) == keys_.end()) {
+      keys_[participant_id] = std::vector<std::vector<uint8_t>>();
+    }
+
+    webrtc::MutexLock lock(&mutex_);
```

Comment:
Same here?

Reply:
> This is 🔥 Just curious, why you didn't use [webrtc::FrameEncryptorInterface](https://github.com/webrtc-sdk/webrtc/blob/m104_release/api/crypto/frame_encryptor_interface.h) (I guess you have not enough flexibility with it?)

Yes, FrameEncryptorInterface is missing the key information of FrameCrypto, codec type (VP8, H264 use different encryption methods), frame type, ssrc and timestamp (for building IV)